src/modes/insert.rs

```diff
@@ -157,6 +171,28 @@ impl Mode for Insert {
                     let delta = ops::delete_cursor(&buffer.data, &buffer.selection);
                     ModeTransition::DirtyBytes(buffer.apply_incomplete_delta(delta))
                 }
+                Action::Move(direction) => {
+                    // we should not move if user already write a half of the hex byte
```

Comment:
Hex half situations are annoying edge cases. Here, I would also find it acceptable if the second half would be set to `0`, and the cursor still moved away. What are your thoughts?

Reply:
:thinking: good point

Reply:
yes, I think inserting a zero is a better approach